client/python/rappor.py

```diff
@@ -119,39 +119,37 @@ def from_csv(f):
     return p
 
 
-class _SimpleRandom(object):
+class _SecureRandom(object):
   """Returns an integer where each bit has probability p of being 1."""
 
-  def __init__(self, prob_one, num_bits, _rand=None):
+  def __init__(self, prob_one, num_bits):
     self.prob_one = prob_one
     self.num_bits = num_bits
-    self._rand = _rand or random.Random()
 
   def __call__(self):
     p = self.prob_one
-    rand_fn = self._rand.random  # cache it for speed
+    bytes = os.urandom(self.num_bits)
 
     r = 0
-    for i in xrange(self.num_bits):
-      bit = rand_fn() < p
+    for i, b in enumerate(xrange(self.num_bits)):
+      bit = b < p * 255.0
```

Comment:
1. Good.
2. >  Do we really need to specify values of P and Q to more than one part in 256?
We don't but enforcing this may be more difficult than trying to respect the parameters as they are passed down to us. Silently rounding Ps and Qs to a multiple of 1/256 is even worse than failing with an error. For example, the difference between using P = .4 and rounding it to the closest multiple of 1/256 (0.3984375) becomes statistically significant after ~40K samples (which would be individual bits in this case). Introducing an avoidable source of error is something we can do without.